### PR TITLE
Victoria/initiallayout: Create initial layout of the App

### PR DESCRIPTION
**Pull request by victoriakario**
I have created a login screen and create account screen, plus set up a basic layout for the logged in pages.

The login/create account screens should be presented in the center of tablet and desktop. While the content should justify to the top left on Mobile devices.

Mobile styles will apply up until 687px, at which point the larger versions should apply.

All colors should be consistent to mockup and the input fields, buttons, tabs and links should all react as the per the mockup.

Let me know if you see anything that doesn't work!
Cheers,
Victoria

### CODE REVIEW

**Comment by victoriakario** [comment]
bitmoji

Added Readme update!~ Ooopsie
Changed name from LoginForm to Login because.... it just looked silly to have Form on CreateAccount and we wanted them to be similar.

File: src/App.tsx [hunk 1]
```diff
@@ -1,10 +1,30 @@
+import GlobalStyle from "globalStyles";
+import styled from "styled-components";
+// import { CreateAccount } from "pages/CreateAccount";
+import { HomeLayout } from "pages/HomeLayout";
+// import { Login } from "pages/Login";
```
Comment (MadOgre) [inline src/App.tsx]:
```suggestion
import styled from "styled-components";

import GlobalStyle from "globalStyles";
// import { CreateAccount } from "pages/CreateAccount";
import { HomeLayout } from "pages/HomeLayout";
// import { Login } from "pages/Login";
```

It's good to separate imports from external libraries and imports from the app itself using a blank line
Also kudos for proper alphabetization

File: src/components/AuthContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import styled from "styled-components";
+import { Logo } from "components/Logo";
+import { ReactNode } from "react";
```
Comment (MadOgre) [inline src/components/AuthContainer.tsx]:
```suggestion
import { ReactNode } from "react";
import styled from "styled-components";

import { Logo } from "components/Logo";
```
Same as above

File: src/components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import styled, { css } from "styled-components";
+import { headingSmall } from "components/Typography";
```
Comment (MadOgre) [inline src/components/Button.tsx]:
```suggestion
import styled, { css } from "styled-components";

import { headingSmall } from "components/Typography";
```
ditto

File: src/components/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+interface ButtonProps {
+  $variant?: "primary" | "secondary";
+}
+
```
Comment (MadOgre) [inline src/components/Button.tsx:8]:
```suggestion
```
Probably don't need extra spacing

File: src/components/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  ${({$variant = "primary"}): typeof secondaryStyles | null => {
+    return $variant === "secondary" ? secondaryStyles : null;
+  }}
+
+  &:disabled {
+    opacity: .25;
+    cursor: not-allowed;
+  }
+`;
```
Comment (MadOgre) [inline src/components/Button.tsx]:
```suggestion
const conditionallyAddSecondaryStyle = ({$variant = "primary"}): typeof secondaryStyles | null => {
  return $variant === "secondary" ? secondaryStyles : null;
}

export const Button = styled.button<ButtonProps>`
  ${headingSmall};
  align-items: center;
  background-color: var(--accent-color);
  border: 0px solid transparent;
  border-radius: 8px;
  color: var(--light-text-color);
  cursor: pointer;
  display: flex;
  flex-direction: row;
  gap: .5rem;
  justify-content: center;
  margin-bottom: 1.5rem;
  outline: none;
  padding: 11px 27px;
  width: 100%;

  &:active:not(:disabled) {
    background-color: var(--accent-hover-color);
    box-shadow: var(--box-shadow-primary);
  }

  ${conditionallyAddSecondaryStyle}

  &:disabled {
    opacity: .25;
    cursor: not-allowed;
  }
`;
```

This function would look much better extracted IMO

File: src/components/InputField.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
+import { useCallback } from "react";
+import styled from "styled-components";
+import { bodyText, bodyTextSmall } from "./Typography";
+import * as icons from "components/icons";
```
Comment (MadOgre) [inline src/components/InputField.tsx]:
```suggestion
import { useCallback } from "react";
import styled from "styled-components";

import * as icons from "components/icons";
import { bodyText, bodyTextSmall } from "components/Typography";
```

File: src/components/InputField.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+  outline: none;
+  padding: 0;
+  &:focus {
+    outline: none;
+  }
+  flex: 1;
+  &::placeholder {
+    opacity: 0.5;
+  }
+`;
```
Comment (MadOgre) [inline src/components/InputField.tsx]:
```suggestion
const Input = styled.input`
  ${bodyText};
  border: none;
  flex: 1;
  outline: none;
  padding: 0;
  &:focus {
    outline: none;
  }
  &::placeholder {
    opacity: 0.5;
  }
`;
```
alphabetization - probably my bad

File: src/components/InputField.tsx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+const Messages = styled.div`
+  ${bodyTextSmall};
+  color: var(--alert-color);
+  flex: 0 0 auto;
+  text-align: right;
+`;
+
```
Comment (MadOgre) [inline src/components/InputField.tsx:73]:
```suggestion
```
extraneous spacing maybe

File: src/components/InputField.tsx [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+      return null;
+    }
+
+    const Icon = styled(icons[icon])`
+      flex: 0 0 auto;
+    `;
+
+    return <Icon />;
+  }, [icon]);
+
```
Comment (MadOgre) [inline src/components/InputField.tsx:86]:
```suggestion
```
too many blank lines

File: src/components/Logo.tsx [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import styled from "styled-components";
+import { ReactComponent as LogoLarge } from "assets/images/logo-devlinks-large.svg";
+import { ReactComponent as LogoSmall } from "assets/images/logo-devlinks-small.svg";
```
Comment (MadOgre) [inline src/components/Logo.tsx]:
```suggestion
import styled from "styled-components";

import { ReactComponent as LogoLarge } from "assets/images/logo-devlinks-large.svg";
import { ReactComponent as LogoSmall } from "assets/images/logo-devlinks-small.svg";
```

File: src/globalStyles.ts [hunk 11]
```diff
@@ -0,0 +1,50 @@
+import { createGlobalStyle } from "styled-components";
+import InstrumentSans from "assets/fonts/InstrumentSans-VariableFont_wdthwght.ttf";
+import InstrumentSansItalic from "assets/fonts/InstrumentSans-Italic-VariableFont_wdthwght.ttf";
```
Comment (MadOgre) [inline src/globalStyles.ts]:
```suggestion
import { createGlobalStyle } from "styled-components";

import InstrumentSans from "assets/fonts/InstrumentSans-VariableFont_wdthwght.ttf";
import InstrumentSansItalic from "assets/fonts/InstrumentSans-Italic-VariableFont_wdthwght.ttf";
```

File: src/globalStyles.ts [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+
+    --box-shadow-primary: 0px 0px 32px 0px rgba(99 60 255 / 25%);
+  }
+
+  * {
+    -webkit-tap-highlight-color: transparent;
+    text-decoration: none;
+    user-select: none;
+  }
+
```
Comment (MadOgre) [inline src/globalStyles.ts:37]:
```suggestion
```
extra line

File: src/index.tsx [hunk 13]
```diff
@@ -1,7 +1,11 @@
 import React from "react";
 import ReactDOM from "react-dom/client";
+import "./polyfills";
 import App from "./App";
 
+import "normalize.css";
+
```
Comment (MadOgre) [inline src/index.tsx]:
```suggestion
import "normalize.css";
import React from "react";
import ReactDOM from "react-dom/client";

import App from "./App";
import "./polyfills";
```
normalize.css is actually a library import

File: src/pages/CreateAccount.tsx [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+  ${bodyText};
+  display: block;
+  margin-bottom: 2.5rem;
+`;
+
+const HelperTextSmall = styled(HelperText)`
+  ${bodyTextSmall};
+  margin-bottom: 1.5rem;
+`;
+
```
Comment (MadOgre) [inline src/pages/CreateAccount.tsx:27]:
```suggestion
```
extra blank line

File: src/pages/CreateAccount.tsx [hunk 15]
```diff
@@ -0,0 +1,92 @@
+import styled from "styled-components";
+
+import { bodyText, bodyTextSmall, headingMedium } from "components/Typography";
+import { InputField } from "components/InputField";
+import { Button } from "components/Button";
+import { AuthContainer } from "components/AuthContainer";
```
Comment (MadOgre) [inline src/pages/CreateAccount.tsx]:
```suggestion
import styled from "styled-components";

import { AuthContainer } from "components/AuthContainer";
import { Button } from "components/Button";
import { InputField } from "components/InputField";
import { bodyText, bodyTextSmall, headingMedium } from "components/Typography";
```
alphabetization

File: src/pages/CreateAccount.tsx [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+
+const HelperTextSmall = styled(HelperText)`
+  ${bodyTextSmall};
+  margin-bottom: 1.5rem;
+`;
+
+
+const CenteredHelperText = styled(HelperText)`
+  text-align: center;
+  margin-bottom: unset;
```
Comment (MadOgre) [inline src/pages/CreateAccount.tsx]:
```suggestion
  margin-bottom: unset;
  text-align: center;
```
m before t :)

File: src/polyfills.js
Comment (MadOgre) [inline src/polyfills.js:1]: Leaving the attribution in - that's mighty big of ya :)

File: src/pages/Login.tsx [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+
+const Heading = styled.div`
+  ${headingMedium};
+`;
+
+const HelperText = styled.div`
+  ${bodyText};
+  display: block;
+  margin-bottom: 2.5rem;
+
```
Comment (MadOgre) [inline src/pages/Login.tsx]:
```suggestion
```
extra blank line

File: src/globalStyles.ts [hunk 18]
```diff
@@ -0,0 +1,51 @@
+import { createGlobalStyle } from "styled-components";
+
+import InstrumentSans from "assets/fonts/InstrumentSans-VariableFont_wdthwght.ttf";
+import InstrumentSansItalic from "assets/fonts/InstrumentSans-Italic-VariableFont_wdthwght.ttf";
+
+const GlobalStyle = createGlobalStyle`
+
```
Comment (MadOgre) [inline src/globalStyles.ts]:
```suggestion
```
blank line

File: src/globalStyles.ts [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+  html {
+    height: 100vh;
+  }
+
+  body {
+    font-family: 'Instrument Sans', sans-serif;
+    font-size: 16px;
+  }
+
```
Comment (MadOgre) [inline src/globalStyles.ts]:
```suggestion
```
blank line

**Review by MadOgre — APPROVED** [review]
Overall great job. Mostly extremely minor nitpicks regarding ordering of and capitalization

File: src/App.tsx (hunk 1 above)
Comment (victoriakario) [inline src/App.tsx]: Thanks! Did this.

File: src/components/Button.tsx (hunk 5 above)
Comment (victoriakario) [reply src/components/Button.tsx]: Wow that does look so much better!!

[bitmoji](https://sdk.bitmoji.com/render/panel/8af413ed-694d-413f-93d2-aee2038b1249-55b80ccd-07ab-4f76-b0c1-7ddd60608ffa-v1.png?transparent=1&palette=1&width=246)

File: src/polyfills.js
Comment (victoriakario) [reply src/polyfills.js:1]: Pay it forward. Someone did the legwork.

**Pull request merged**